src/collective/collectionfilter/vocabularies.py

```diff
@@ -100,6 +101,9 @@ def translate_portal_type(value):
     """Translate the type based on its i18n domain the fti provides."""
     types_tool = plone.api.portal.get_tool('portal_types')
     fti = {}
+    if type(value) == Message:
```

Comment:
why is the value a `Message` object? doesn't make sense ...

Reply:
because this is what i get there
```python
type(value)
<type 'zope.i18nmessageid.message.Message'>
```

i don't get that either.
and secondly: as "Page" is a message i get, this is not represented in the ReallyUserFriendlyTypesVocabulary.

there are two selectable values: "portal_type" or "Type"

Reply:
this "page" is not translated:
```python
(Pdb) vocabulary.getTermByToken('Page').title
'Page'
```
while
```python
(Pdb) vocabulary.getTermByToken('Document').title
u'Seite'
```
is

Reply:
when switching to "portal_type" in the group_by settings of the filter portlet, i get it as you mentioned - using the vocabulary. which is fine ;)

but on the other value "Type" i get the "wrong" type name, which is not part of the vocab and therefore no translation.

Reply:
well the index `Type` is the `Title` of the FTI ... see https://github.com/plone/plone.dexterity/blob/master/plone/dexterity/content.py#L491 ... IMHO this is only for historical reasons ... never use `Type` for filtering. The solution is that collectionfilter only translates `portal_type` index while `Type` index is up to the integrator what to do with it

Reply:
so this is the answer, why you get a `Message` object as value, because the `Type` index holds the title of the FTI ... lesson learned

Reply:
as i can select "Type" besides "portal_type" i want them to work as intended - if the index "Type" might be gone some day, i'm happy to wave him good bye. till now i need to take care of that. ;)

Reply:
i refactored it entirely and separated this to two methods "translate_type" and "translate_portal_type"

hope this is more elegant (if we can speak of elegance on this thingy)

Reply:
resolved?

Reply:
see my changes: it's simply enough to use `translate` for `brain.Type` as its already a MessageFactory ...

Reply:
ok - i didn't see that coming :) thank you for your work!